urllib3/util/selectors.py

```diff
@@ -112,10 +117,9 @@ def _syscall_wrapper(func, recalc_timeout, *args, **kwargs):
                     if expires is not None:
                         current_time = monotonic()
                         if current_time > expires:
-                            raise OSError(errno=errno.ETIMEDOUT)
+                            raise OSError(errno.ETIMEDOUT, "Connection timed out")
                         if recalc_timeout:
-                            if "timeout" in kwargs:
-                                kwargs["timeout"] = expires - current_time
+                            kwargs["timeout"] = expires - current_time
```

Comment:
That's true, I'm wondering how this wasn't caught by testing. I know timeout testing is disabled on CI but I should have seen this locally. Might have to write more tests.